IMDA-BTG/aiverify
# added codes for fairness for image testing

## Description

Support for image fairness testing 

## Motivation and Context

Support for image fairness testing 

## Type of Change

 feat: A new feature

## How to Test

[Provide clear instructions on how to test and verify the changes introduced by this pull request, including any specific unit tests you have created to demonstrate your changes.]
1) Create a new project and drag a new FMTC bar chart to the canvas and continue
2) In stock-plugins/aiverify.stock.fairness-metrics-toolbox-for-classification/algorithms/fairness_metrics_toolbox_for_classification/tests/user_defined_files, upload and use:
    - data/small_test as Testing Dataset
    - data/pickle_pandas_annotated_labels_50.sav as Ground Truth Dataset
    - pipeline/bc_image_face_1
3) Set **Ground Truth** to **gender**
4) Set 
     - **Test Arguments Sensitive Feature Names** to **race**, 
     - **Annotated** labels **path** to **pickle_pandas_annotated_labels_50.sav** 
     - **Column name** to **image_directory**
5) Fill in the input blocks with any value until you see the fairness tree and click OK
6) Run the algorithm

## Checklist
Please check all the boxes that apply to this pull request using "x":


## Screenshots (if applicable)

![fmtc_image_testing](https://github.com/IMDA-BTG/aiverify/assets/129382289/2357b752-fce4-4c48-b596-a429005c360b)

## Additional Notes

-

<details>
  <summary>Developer Certificate of Origin</summary>

 ```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1


By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
 ```
</details>
